### PR TITLE
[WIP] Convert HelperFunctions to ts

### PR DESCRIPTION
**Pull request by hydroflame**

### CODE REVIEW

File: utils/HelperFunctions.ts [hunk 1]
```diff
@@ -22,54 +13,61 @@ function clearObject(obj) {
...
+function removeElementById(id: string) {
+    const elem: HTMLElement | null = document.getElementById(id);
+    if(elem === null) return;
+    removeElement(elem as HTMLElement);
 }
 
-function removeElement(elem) {
-    if (elem == null || !(elem instanceof Element)) {return;}
+// Same as removeElementById except if takes a DOM element object rather than an ID
+function removeElement(elem: Element) {
```
Comment (kopelli) [inline utils/HelperFunctions.ts]: why the use of "Element" vs. "HTMLElement"? Usage seems to be mixed. I'm sure there is a hierarchy between them, just not sure which is the more general and if we really need it to be.

File: utils/HelperFunctions.ts [hunk 2]
```diff
@@ -80,17 +78,57 @@ function removeChildrenFromElement(el) {
...
+    tabIndex?: any;
+    clickListener?: any;
+    inputListener?: any;
+    changeListener?: any;
+    onkeyup?: any;
+    onfocus?: any;
+}
+
+function createElement(type: string, params: createElementParams): any {
+    var el: any = document.createElement(type);
```
Comment (kopelli) [inline utils/HelperFunctions.ts]: does `Element` or `HTMLElement` not work?

File: utils/HelperFunctions.ts [hunk 3]
```diff
@@ -165,19 +203,23 @@ function createPopup(id, elems) {
...
 
-    for (var i = 0; i < elems.length; ++i) {
+    for (let i = 0; i < elems.length; ++i) {
         content.appendChild(elems[i]);
     }
     container.appendChild(content);
-    document.getElementById("entire-game-container").appendChild(container);
+    const game = document.getElementById("entire-game-container");
+    if(game === null) {
+        throw new Error('unable to find entire-game-container, massive problem');
```
Comment (kopelli) [inline utils/HelperFunctions.ts]: 👍 We don't make use of true errors enough.

**Comment by danielyxie** [comment]
Done in #352

**Pull request closed without merging**